src/GHC/SourceGen/Decl.hs

```diff
@@ -306,10 +318,10 @@ data' = newOrDataType DataType
 --
 -- > Foo a Int
 -- > =====
--- > conDecl "Foo" [field (var "a"), field (var "Int")]
+-- > prefixCon "Foo" [field (var "a"), field (var "Int")]
 prefixCon :: OccNameStr -> [Field] -> ConDecl'
 prefixCon name fields = renderCon98Decl name
-    $ PrefixCon $ map renderField fields
+    $ PrefixCon $ map (hsUnrestricted . renderField) fields
```

Comment:
This serves as a kind of to-do.

GHC 9 requires this change because it introduces two variants: `hsUnrestricted` and `hsLinear` for linear arrows. in the future, it will be desirable to actually support those features, but for now I just reproduced the old behavior.

cc @judah